src/main/src/main/java/org/geoserver/catalog/impl/CascadedLayerInfoImpl.java

```diff
@@ -0,0 +1,254 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.geoserver.catalog.CascadedLayerInfo;
+import org.geoserver.catalog.LayerInfo;
+import org.geoserver.catalog.LegendInfo;
+import org.geoserver.catalog.StyleInfo;
+import org.geoserver.catalog.WMSLayerInfo;
+import org.geotools.ows.wms.StyleImpl;
+import org.geotools.styling.NamedStyleImpl;
+import org.geotools.styling.Style;
+import org.geotools.util.logging.Logging;
+
+public class CascadedLayerInfoImpl extends LayerInfoImpl implements CascadedLayerInfo {
+
+    static final Logger LOGGER = Logging.getLogger(CascadedLayerInfo.class);
+
+    /** serialVersionUID */
+    private static final long serialVersionUID = -4772580517514254546L;
+
+    protected String forcedRemoteStyle = "";
+    protected String prefferedFormat = "image/png";
+
+    private List<String> selectedRemoteFormats = new ArrayList<String>();
+
+    private List<String> selectedRemoteStyles = new ArrayList<String>();
+
+    public CascadedLayerInfoImpl() {}
+
+    public CascadedLayerInfoImpl(LayerInfo delegate) {
+
+        super.setResource(delegate.getResource());
+        super.setName(delegate.getResource().getName());
+        super.setEnabled(delegate.enabled());
+        super.setType(delegate.getType());
+    }
+
+    @Override
+    public void reset() {
+        // select all formats for use
+        selectedRemoteStyles.addAll(getRemoteStyles());
+        // set first style is selected
+        if (!selectedRemoteStyles.isEmpty()) forcedRemoteStyle = selectedRemoteStyles.get(0);
```

Comment:
Is true that WMS capabilities documents don't allow the definition of a default style, so picking up the first one is the good approach:

> If the server advertises several styles for a layer, and the client sends a request for the default style, the choice of
which style to use as default is at the discretion of the server. The ordering of styles in the service metadata does
not indicate which is the default.

Reply:
Actually the real default is to send _""_, and that's what we shoudl be doing.

Reply:
understood

Reply:
You could have an empty option, meaning use the default, would that make sense?

Reply:
agreed